src/FSharp.Editing.Messages/Messages.fs

```diff
@@ -0,0 +1,46 @@
+﻿namespace FSharp.Editing.Messages
+
+type [<Measure>] requestId
+type RequestId = int64<requestId>
+type FilePath = string
+type DocumentContent = string
+
+type Project =
+    { FileName: FilePath
+      Files: FilePath list
+      ReferencedProjects: FilePath list
+      CompilerOptions: string list }
+
+type ServerCapability =
+    | SyntaxHighlighting
+    | HighlighIdentifier
+    | ImplementInterface
+
+type ClientNotification =
+      /// New solution loaded or new folder opened. All server state should be discarded.
+    | Cleanup
+      /// A new project loaded or a known project changed in any way (name, file list, references, etc.)
+    | ProjectLoaded of Project
+      /// Previously loaded project was unloaded from IDE or deleted from disk so it should not be part of solution anymore.
+    | ProjectUnloaded
```

Comment:
`ProjectUnloaded` should contain some data saying which project to unload. I guess path would be enough